sdk/provisioning/Azure.Provisioning/src/resources/DeploymentScript.cs

```diff
@@ -61,11 +62,11 @@ public DeploymentScript(IConstruct scope, string resourceName, Resource database
                 cleanupPreference: ScriptCleanupOptions.OnSuccess,
                 environmentVariables: new List<ScriptEnvironmentVariable>
                 {
+                    new ScriptEnvironmentVariable("APPUSERPASSWORD"),
+                    new ScriptEnvironmentVariable("SQLCMDPASSWORD"),
+                    new ScriptEnvironmentVariable("DBSERVER"),
+                    new ScriptEnvironmentVariable("DBNAME") { Value = database.Id.Name },
```

Comment:
I think this accomplishes the same thing as grabbing the "name" property from resource name reference.